k4MarlinWrapper/examples/event_display.py

```diff
@@ -18,58 +18,71 @@
 # limitations under the License.
 #
 
-from Gaudi.Configuration import *
+import os
+import sys
+
+from Gaudi.Configuration import INFO
+from Configurables import MarlinProcessorWrapper, k4DataSvc, PodioInput, EDM4hep2LcioTool, LcioEvent, GeoSvc
+from k4FWCore.parseArgs import parser
+
+parser.add_argument(
+    "--inputFiles",
+    action="extend",
+    nargs="+",
+    metavar=["file1", "file2"],
+    help="One or multiple input files",
+)
+parser.add_argument(
+    "--compactFile", help="Compact detector file to use", type=str, default=""
+)
+
+reco_args = parser.parse_known_args()[0]
 
-from Configurables import MarlinProcessorWrapper, k4DataSvc, PodioInput, EDM4hep2LcioTool
 algList = []
+svcList = []
+
+evtsvc = k4DataSvc("EventDataSvc")
+svcList.append(evtsvc)
+
+if reco_args.compactFile:
+    compact_file = reco_args.compactFile
+else:
+    # add the same default as before
+    # TODO: maybe modernize this
+    compact_file = "CLICPerformance/Visualisation/CLIC_o3_v06_CED/CLIC_o3_v06_CED.xml"
```

Comment:
This could be handled via `default=<...>` in the corresponding `add_argument` call. Unless we make it a required argument and then `argparse` will error out for us if there is none. I actually like that solution best, because what good is a default geometry for an event display that is used by different communities?

Reply:
I am all for having to explicitly pass the detector model.

 I tried to keep some of the original behaviour when I wrote this but as I later on noticed that I need to rewrite the documentation anyway we can also change it now :)